### ISSUE TITLE
Align flow-parser with babylon v7

### ISSUE THREAD

**Issue by danez** [facebook/flow#4789]
Hi,

we are getting close to release a beta of v7 of babel soon.

We changed some things in v7 in coordination with you guys, and this issue is a reminder to integrate this changes into flow:

### Create separate Variance Node
As coordinated with @samwgoldman the variance should be represented as separate node, which is already done in babel v7. Flow should be updated accordingly
Related tickets: babel/babel#4722, babel/babel#4697, babel/babylon#161, babel/babylon#333

### Add method property to FunctionTypeAnnotation node
This was discussed and seems agreed on. babylon has an open PR for this: babel/babylon/334
Related tickets: babel/babylon#331

I would create PRs for this myself, but well OCaml :)

Thanks.

//cc @samwgoldman @mroch

**Comment by mroch** [comment]
Method property added in https://github.com/facebook/flow/commit/622bbc4f07acb77eb1109830c70815f827401d90

@danez It looks like babel/babylon#331 became babel/babel#6704 and is still open. its PR was closed without being merged.

**Comment by danez** [comment]
Thanks

**Comment by samwgoldman** [comment]
I have a change pending review internally for the variance node. Just waiting on @mroch to review :)
